src/engine/guidance/post_processing.cpp

```diff
@@ -497,11 +499,43 @@ void collapseTurnAt(std::vector<RouteStep> &steps,
     }
 }
 
+// Works on steps including silent and invalid instructions in order to do lane anticipation for
+// roundabouts which later on get collapsed into a single multi-hop instruction.
+std::vector<RouteStep> anticipateLaneChangeForRoundabouts(std::vector<RouteStep> steps)
+{
+    util::guidance::print(steps);
+
+    const constexpr auto MIN_DURATION_NEEDED_FOR_LANE_CHANGE_INSIDE_ROUNDABOUT = 15.;
+
+    using StepIter = decltype(steps)::iterator;
+    using StepIterRange = std::pair<StepIter, StepIter>;
+
+    const auto anticipate_lanes_in_quick_roundabouts = [&](StepIterRange roundabout) {
+        // We only care about roundabouts where there is no time to change lanes inside
+        const auto sum = [](const double v, const RouteStep &step) { return v + step.duration; };
+        const auto seconds_inside = std::accumulate(roundabout.first, roundabout.second, 0., sum);
+
+        if (seconds_inside >= MIN_DURATION_NEEDED_FOR_LANE_CHANGE_INSIDE_ROUNDABOUT)
+            return;
+
+        // We do lane anticipation on the roundabout's enter and leave step only.
+        // TODO: This means, lanes _inside_ the roundabout are ignored at the moment.
```

Comment:
only lanes on the enter instruction are reported to the outside. This paradigm has to be reworked, if it should work in the end.